include/netcdf.h

```diff
@@ -293,11 +293,13 @@ NOTE: The NC_MAX_DIMS, NC_MAX_ATTRS, and NC_MAX_VARS limits
 
 /** In HDF5 files you can set storage for each variable to be either
  * contiguous or chunked, with nc_def_var_chunking().  This define is
- * used there. */
+ * used there. Unknown storage is used for further extensions of HDF5
+ * storage models, which should be handled transparently by netcdf */
 /**@{*/
-#define NC_CHUNKED    0
-#define NC_CONTIGUOUS 1
-#define NC_COMPACT    2
+#define NC_CHUNKED         0
+#define NC_CONTIGUOUS      1
+#define NC_COMPACT         2
+#define NC_UNKNOWN_STORAGE 3
```

Comment:
Can we call this NC_VIRTUAL instead?

Reply:
We could, but my line of thinking was that we maybe should first add a default case, which should be able to handle all possible storage methods of HDF5 in a generic, but maybe less efficient way. This would future-proof the implementation if even more storage mechanisms would be included into HDF5.

`NC_VIRTUAL` would then be an additional way of handling storage, which could enable further optimizations which are only applicable to virtual datasets. We could add that now or whenever it is actually needed.

Reply:
OK, I think you should add NC_VIRTUAL.

Are there any other storage types besides chunked, contiguous, compact, and virtual?

Reply:
OK, I get it now. NC_UNKNOWN is good. I think you should add NC_VIRTUAL and have them both.

Reply:
Now we have both, I think this a good solution. Also `ncdump` is now at least able to tell that it is a virtual dataset.
I've been a bit reluctant to change `NC_UNKNOWN_STORAGE` into `NC_UNKNOWN` as the term is quite generic and I am afraid that it might collide with other things at some point in time. But I also don't have a strong preference, so if you like `NC_UNKNOWN` more then I'll change that as well.

Reply:
No NC_UNKNOWN_STORAGE is good.